src/miner_keys.erl

```diff
@@ -28,7 +30,14 @@ keys({file, BaseDir}) ->
                key_slot => undefined,
                ecdh_fun => libp2p_crypto:mk_ecdh_fun(PrivKey0),
                sig_fun => libp2p_crypto:mk_sig_fun(PrivKey0),
-               onboarding_key => undefined
+               onboarding_key => case filelib:ensure_dir(OnboardingKey) of
```

Comment:
Why are you trying to create the directory. Check if it's there, don't ensure its creation .

In fact only use the file:read_file to load it or fail?

Reply:
I thought `filelib:ensure_dir` only checked for the existence of a directory. I am newb.